### PR TITLE
CASTable docstring edits

### PR DESCRIPTION
**Pull request by kybowm**
Here are some changes to CASTable docstrings. There were no changes to functional code.

None of the docstring edits are critical, but since the CASTable API is absorbed by other products like dlpy.ImageTable, I wanted to make them stylistically similar. Style is much less important than function, though. If these suggestions don't effectively communicate how swat works, I'm open to changes.

### CODE REVIEW

File: swat/cas/table.py [hunk 1]
```diff
@@ -2784,7 +2783,7 @@ def head(self, n=5, columns=None, bygroup_as_index=True, casout=None):
         columns : list-of-strings, optional
             A subset of columns to return.
         bygroup_as_index : boolean
-            If By groups are specified, should they be converted to an index?
+            When By groups are specified, if bygroup_as_index is True, By groups are converted to an index
```
Comment (kesmit13) [inline swat/cas/table.py]: I'm not fond of the wording here.  Maybe "When By groups are specified and bygroup_as_index is True, By groups are converted to an index.  Also, lines should be kept to <78 characters.

File: swat/cas/table.py [hunk 2]
```diff
@@ -2813,7 +2812,7 @@ def tail(self, n=5, columns=None, bygroup_as_index=True, casout=None):
...
```
Comment (kesmit13) [inline swat/cas/table.py]: Same as above

File: swat/cas/table.py [hunk 3]
```diff
@@ -2846,7 +2845,7 @@ def slice(self, start=0, stop=None, columns=None, bygroup_as_index=True, casout=
...
```
Comment (kesmit13) [inline swat/cas/table.py]: Same as above

File: swat/cas/table.py [hunk 4]
```diff
@@ -3425,7 +3424,7 @@ def _materialize(self, casout=None, inplace=False, prefix=None, suffix=None):
         casout : dict, optional
             The CAS output table definition
         inplace : bool, optional
-            Should the output table be overwritten?
+            If True, the output table is overwritten
```
Comment (kesmit13) [inline swat/cas/table.py]: You might want to add "if it exists" at the end

File: swat/cas/table.py [hunk 5]
```diff
@@ -3506,7 +3505,7 @@ def _bool(self):
 
     def all(self, axis=None, bool_only=None, skipna=True, level=None, **kwargs):
         '''
-        Return whether all elements in the column are True
+        Return True for each column with only True elements
```
Comment (kesmit13) [inline swat/cas/table.py]: This wording makes it sound like the items have to literally be True values, but it's really any value that is treated as true.  Maybe just not capitalize the second "True".

File: swat/cas/table.py [hunk 6]
```diff
@@ -3555,7 +3554,7 @@ def all(self, axis=None, bool_only=None, skipna=True, level=None, **kwargs):
 
     def any(self, axis=None, bool_only=None, skipna=True, level=None, **kwargs):
         '''
-        Return whether any elements in the column are True
+        Return True for each column with at least one True element
```
Comment (kesmit13) [inline swat/cas/table.py]: Don't capitalize second "True".

File: swat/cas/table.py [hunk 7]
```diff
@@ -7798,20 +7796,20 @@ def capitalize(self):
 
     def contains(self, pat, case=True, flags=0, na=np.nan, regex=True):
         '''
-        Does the value contain the specified pattern?
+        Indicates wheter the value contains the specified pattern.
```
Comment (kesmit13) [inline swat/cas/table.py]: Mis-spelled "whether"

Comment (kesmit13) [inline swat/cas/table.py]: No period at end.  This should be done for all of the one-line descriptions of methods.

File: swat/cas/table.py [hunk 8]
```diff
@@ -7865,20 +7863,20 @@ def count(self, pat, flags=0, **kwargs):
 
     def endswith(self, pat, case=True, flags=0, na=np.nan, regex=True):
         '''
-        Does the table column end with the given pattern?
+        Indicates whether the table column ends with the given pattern.
```
Comment (kesmit13) [inline swat/cas/table.py]: No period at end

File: swat/cas/table.py [hunk 9]
```diff
@@ -8176,20 +8174,20 @@ def slice(self, start=0, stop=None, step=None):
 
     def startswith(self, pat, case=True, flags=0, na=np.nan, regex=True):
         '''
-        Does the table column start with the given pattern?
+        Indicates wheter the table column start with the given pattern.
```
Comment (kesmit13) [inline swat/cas/table.py]: Mis-spelled "whether"

File: swat/cas/table.py [hunk 10]
```diff
@@ -8283,7 +8281,7 @@ def isalpha(self):
 
     def isdigit(self):
         '''
-        Does the value contain only digits?
+        Indicates wheter the value contains only digits.
```
Comment (kesmit13) [inline swat/cas/table.py]: "whether"

File: swat/cas/table.py [hunk 11]
```diff
@@ -8328,7 +8326,7 @@ def islower(self):
 
     def isupper(self):
         '''
-        Does the value contain only uppercase characters?
+        Does the value contains only uppercase characters.
```
Comment (kesmit13) [inline swat/cas/table.py]: Put question mark back in.  The description is phrased as a question.

File: swat/cas/table.py (hunk 1 above)
Comment (kybowm) [reply swat/cas/table.py]: When by_group_index is True, By groups are converted to an index if they exist

File: swat/cas/table.py (hunk 2 above)
Comment (kybowm) [reply swat/cas/table.py]: Same as above

File: swat/cas/table.py (hunk 3 above)
Comment (kybowm) [reply swat/cas/table.py]: Same as above

File: swat/cas/table.py (hunk 4 above)
Comment (kybowm) [reply swat/cas/table.py]: Changed as suggested.

File: swat/cas/table.py (hunk 5 above)
Comment (kybowm) [reply swat/cas/table.py]: Return True for each column with only elements treated as True

File: swat/cas/table.py (hunk 7 above)
Comment (kybowm) [reply swat/cas/table.py]: Fixed

Comment (kybowm) [reply swat/cas/table.py]: Fixed for all.

File: swat/cas/table.py (hunk 8 above)
Comment (kybowm) [reply swat/cas/table.py]: Fixed.

File: swat/cas/table.py (hunk 9 above)
Comment (kybowm) [reply swat/cas/table.py]: Fixed

File: swat/cas/table.py (hunk 10 above)
Comment (kybowm) [reply swat/cas/table.py]: Fixed

File: swat/cas/table.py (hunk 6 above)
Comment (kybowm) [reply swat/cas/table.py]: Return True for each column with at least one element treated as True

**Comment by kybowm** [comment]
@kesmit13 I forgot I had this PR open. I believe I have addressed all your suggestions. I changed many as you suggested, but for some I proposed alternatives. I also updated my fork with the recent changes to swat. Let me know what you think of the edits.

File: swat/cas/table.py (hunk 6 above)
Comment (kesmit13) [reply swat/cas/table.py:3557]: I'm still not fond of the second literal 'True' here.  The values don't have to literally be 'True', they just have to be true by rules of truth (i.e., true == non-zero / non-empty-string / etc.).

Comment (kybowm) [reply swat/cas/table.py]: Okay. I changed the second 'True' to 'true' in any() and all().

File: swat/cas/table.py [hunk 12]
```diff
@@ -126,7 +126,7 @@ def _get_unique(seq, lowercase=False):
     Parameters
     ----------
     lowercase : boolean
-        Should values be compared in a case-insensitive way?
+        If true, compare elements in a case-insensitive way
```
Comment (kesmit13) [inline swat/cas/table.py:129]: This 'true' should be capitalized to match the other changes.  After this change, I'm fine with the whole set of commits.

Comment (kybowm) [reply swat/cas/table.py]: Sounds good. I just updated it. Thanks!

**Pull request merged**